### ISSUE TITLE
Plural rules are broken for time diff strings

### ISSUE THREAD

**Issue by TheoChevalier** [mozilla-services/screenshots#3495]
Screenshot of timeDiffHoursAgo in French https://screenshots.firefox.com/T7g56E2IcYW6yQkK/screenshots.firefox.com

According to the translation here, it should be 1 heure (without s) https://pontoon.mozilla.org/fr/firefox-screenshots/server.ftl/?search=timeDiffHoursAgo&string=168527

Flod pointed that there might be a problem here https://github.com/mozilla-services/screenshots/blob/954a54d4ab4c212816e2a18f41ad260cea0e24a3/server/src/pages/shot/time-diff.js#L17

CC: @stasm @flodolo

**Comment by 6a68** [comment]
@stasm Is it intended that passing the number 1 chooses the "one" plural, but passing the string "1" chooses the default, rather than coercing the string to a number?

this behavior is visible in the fluent playground:

"1" as a string:
![](https://screenshots.firefoxusercontent.com/images/07daa838-afd9-4c04-8837-2b467e0eab40.png)

1 as a number:
![](https://screenshots.firefoxusercontent.com/images/82d0e864-c04e-4815-951e-bc24fc40062c.png)

**Comment by stasm** [comment]
Oops, sorry I missed the notification here. I'm happy I didn't block you from fixing this.

To answer @6a68's question: yes, this is the intended behavior for two reasons:

  - It's more flexible that way: developers can choose to `parseInt`. If Fluent coerced, developers wouldn't be able to choose not to.
  - Some locales use different plural forms for integers and decimals, e.g. `1` is different from `1.0`. I think English is one of them: `1 apple` but `1.0 apples`. The only way in JS to pass `1.0` and not lose the decimal part is to pass it as a string. (That said, fluent.js doesn't yet correctly support such numbers.)

**Comment by 6a68** [comment]
@stasm Nah, not a problem. Good point re: decimal singular, I hadn't considered that.
